src/sign/process.rs

```diff
@@ -0,0 +1,216 @@
+        system_instruction, system_program, sysvar,
+        transaction::Transaction,
+    },
+    Client, Program,
+};
+use console::style;
+use futures::future::select_all;
+use mpl_token_metadata::{instruction::sign_metadata, ID as METAPLEX_PROGRAM_ID};
+use retry::{delay::Exponential, retry};
+use solana_client::rpc_client::RpcClient;
+use solana_transaction_crawler::crawler::Crawler;
+use std::{
+    cmp,
+    str::FromStr,
+    sync::{
+        atomic::{AtomicBool, Ordering},
+        Arc,
+    },
+};
+
+use crate::{
+    cache::load_cache,
+    candy_machine::CANDY_MACHINE_ID,
+    common::*,
+    config::SugarConfig,
+    pdas::find_metadata_pda,
+    setup::{setup_client, sugar_setup},
+    utils::*,
+};
+
+pub struct SignArgs {
+    pub candy_machine_id: Option<String>,
+    pub keypair: Option<String>,
+    pub cache: String,
+    pub rpc_url: Option<String>,
+    pub mint: Option<String>,
+    pub interrupted: Arc<AtomicBool>,
+}
+
+pub async fn process_sign(args: SignArgs) -> Result<()> {
+    // (1) Setting up connection
+    println!(
+        "{} {}Initializing connection",
+        if args.mint.is_some() {
+            style("[1/2]").bold().dim()
+        } else {
+            style("[1/3]").bold().dim()
+        },
+        COMPUTER_EMOJI
+    );
+
+    let pb = spinner_with_style();
+    pb.set_message("Connecting...");
+
+    let sugar_config = Arc::new(sugar_setup(args.keypair, args.rpc_url)?);
+
+    let candy_machine_id = match args.candy_machine_id {
+        Some(candy_machine_id) => candy_machine_id,
+        None => {
+            let cache = load_cache(&args.cache, false)?;
+            cache.program.candy_machine
+        }
+    };
+
+    pb.finish_with_message("Connected");
+
+    if let Some(mint_id) = args.mint {
+        println!(
+            "\n{} {}Signing one NFT",
+            style("[2/2]").bold().dim(),
+            SIGNING_EMOJI,
+        );
+        let pb = spinner_with_style();
+        pb.set_message(format!("Signing NFT with mint id {}.", mint_id));
+
+        let account_pubkey = Pubkey::from_str(&mint_id)?;
+        let metadata_pubkey = find_metadata_pda(&account_pubkey);
+
+        match sign(Arc::clone(&sugar_config), metadata_pubkey).await {
+            Ok(signature) => format!("{} {:?}", style("Signature:").bold(), signature),
+            Err(err) => {
+                pb.abandon_with_message(format!("{}", style("Signing failed ").red().bold()));
+                error!("{:?}", err);
+                return Err(err);
+            }
+        };
+
+        pb.finish();
+    } else {
+        println!(
+            "\n{} {}Fetching mint ids",
+            style("[2/3]").bold().dim(),
+            SIGNING_EMOJI,
+        );
+
+        let mut errors = Vec::new();
+        let client = RpcClient::new("https://ssc-dao.genesysgo.net");
+        let cm_id = Pubkey::from_str(&candy_machine_id).unwrap();
+
+        let crawled_accounts = &Crawler::get_cmv2_mints(client, cm_id).await?["metadata"];
```

Comment:
I know this is how my example does it in the `solana-transaction-crawler` library, but it might be better to use `.get("metadata")` instead of indexing directly. The latter will panic if there are no metadata accounts and the former will return a `None`.